apps/backend/src/modules/users/user.repository.ts

```diff
@@ -11,52 +16,199 @@ class UserRepository implements Repository {
 	}
 
 	public async create(entity: UserEntity): Promise<UserEntity> {
-		const { email, firstName, lastName, passwordHash, passwordSalt } =
+		const { email, firstName, groupId, lastName, passwordHash, passwordSalt } =
 			entity.toNewObject();
 
-		const user = await this.userModel
-			.query()
-			.insert({
+		return await transaction(this.userModel, async (UserModel) => {
+			const { id: userId } = await UserModel.query().insert({
 				email,
 				firstName,
+				groupId,
 				lastName,
 				passwordHash,
 				passwordSalt,
-			})
-			.returning("*")
-			.execute();
+			});
 
-		return UserEntity.initialize(user);
+			const user = (await UserModel.query()
+				.where("users.id", userId)
+				.withGraphJoined("group.permissions")
+				.first()) as UserModel;
+
+			const group = user.group as NonNullable<typeof user.group>;
+			const permissions = user.group?.permissions as NonNullable<
+				typeof group.permissions
+			>;
+
+			return UserEntity.initialize({
+				email: user.email,
+				firstName: user.firstName,
+				group: GroupEntity.initializeWithPermissions({
+					id: group.id,
+					key: group.key,
+					name: group.name,
+					permissions: permissions.map((per) =>
```

Comment:
We need to avoid contractions except for very common ones,
```suggestion
					permissions: permissions.map((permission) =>
```